LinkedListNode.swift

```diff
@@ -1,38 +1,56 @@
 
-public class LLNode<T: Equatable & Comparable & CustomStringConvertible> {
+public final class LLNode<T> {
     public var next: LLNode<T>?
     public var value: T
-
-    public init(value: T, next: LLNode<T>? = nil) {
+    
+    public required init(value: T, next: LLNode<T>? = nil) {
         self.value = value
         self.next = next
     }
+}
+
+public protocol Copiable: class {
+    func copy() -> Self
+}
 
-    public func deepCopy() -> LLNode {
-        return LLNode(value: value, next: next?.deepCopy())
+extension LLNode: Copiable where T: Copiable {
+    public func copy() -> LLNode {
+        return LLNode(value: value.copy(), next: next?.copy())
     }
 }
 
-extension LLNode: Equatable {
+extension LLNode: Equatable where T: Equatable {
     public static func ==(lhs: LLNode<T>, rhs: LLNode<T>) -> Bool {
         guard lhs.value == rhs.value else {
             return false
         }
-
+        
         return lhs.next == rhs.next
     }
 }
 
-extension LLNode: Comparable {
-    public static func <(lhs: LLNode<T>, rhs: LLNode<T>) -> Bool {
```

Comment:
I removed this method because I'm not sure what it means to compare one node at a time. The values inside the nodes are what we want to compare, but using this function obscures that.